### PR TITLE
feature: popover for dataset filters

### PR DESCRIPTION
**Pull request by densumesh**
Added:
- Ability to search through datasets
- Select multiple datasets
- Filters added to URL when searching
- Filters pulled down from URL on the results page

### CODE REVIEW

**Comment by densumesh** [comment]
Screenshot 2023-06-05 at 11 44 34 AM
The styling is pretty rough since I'm not familiar with tailwind at all but this is a working feature still.

**Comment by kmkhami** [comment]
Is this a combobox?

File: src/FilterStore.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import { createSignal } from 'solid-js'
+import type { comboboxItem } from './components/Combobox'
+
+export const [selectedComboboxItems, setSelectedComboboxItems] = createSignal<comboboxItem[]>([])
```
Comment (kmkhami) [inline src/FilterStore.ts:4]: Can we move this to be within `Combobox.tsx`? It's typically not worth making a new file for only 4 lines of code.

**Comment by densumesh** [comment]
Yeah, a combobox means that the user can search through the filters through text input right?

File: src/components/Combobox.tsx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+}
+
+
+export const Combobox = () => {
+	const [panelOpen, sePanelOpen] = createSignal(false)
+	const [usingPanel, setUsingPanel] = createSignal(false)
+	const [inputValue, setInputValue] = createSignal('')
+	const [comboboxItems, setComboboxItems] = createSignal<comboboxItem[]>([
+		{ name: 'Policy', eventId: 'policy' },
+		{ name: 'LD', eventId: 'ld' }
```
Comment (kmkhami) [inline src/components/Combobox.tsx:18]: We should be either defining these above line 12 as a static object or pass them as a prop. 

I would prefer to define them as a static object though.

**Comment by densumesh** [comment]
image

File: src/components/SearchForm.tsx [hunk 3]
```diff
@@ -11,20 +12,20 @@ const SearchForm = (props: { query?: string }) => {
...
+		e.preventDefault()
+		const textAreaValue = textareaInput()
+		const searchQuery = encodeURIComponent(
+			textAreaValue.length > 3800 ? textAreaValue.slice(0, 3800) : textAreaValue
+		)
+		const filters = selectedComboboxItems()
+			.map((item) => item.eventId)
+			.join(',')
+		window.location.href = `/search?q=${searchQuery}&filters=${filters}`
+	}
```
Comment (kmkhami) [inline src/components/SearchForm.tsx:25]: Thank you for paying off this tech debt :+1:

File: src/components/SearchForm.tsx [hunk 4]
```diff
@@ -11,20 +12,20 @@ const SearchForm = (props: { query?: string }) => {
...
-
+	const onSubmit = (e: Event) => {
+		e.preventDefault()
+		const textAreaValue = textareaInput()
+		const searchQuery = encodeURIComponent(
+			textAreaValue.length > 3800 ? textAreaValue.slice(0, 3800) : textAreaValue
+		)
+		const filters = selectedComboboxItems()
+			.map((item) => item.eventId)
+			.join(',')
```
Comment (kmkhami) [inline src/components/SearchForm.tsx:23]: Is there a reason not to use encodeUriComponent here?

**Review by kmkhami — CHANGES_REQUESTED** [review]
I still need to pull this down and check the styling.

File: src/components/SearchForm.tsx (hunk 4 above)
Comment (densumesh) [reply src/components/SearchForm.tsx:23]: No I just forgot to put that in, but it shouldn't be necessary as all the filters shouldn't require any special characters.

**Comment by kmkhami** [comment]
I made some changes: 


https://github.com/arguflow/vault/assets/15804464/08e429f2-cfc9-4df4-a34f-9bd74d2741a7

**Review by kmkhami — APPROVED** [review]

**Pull request merged**